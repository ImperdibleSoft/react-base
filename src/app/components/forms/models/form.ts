import { FormState } from '../types/form';

export const getState = (): FormState => ({
  canBeLocked: true,
  hasPermissions: false,
  isEditEnabled: false,
});
