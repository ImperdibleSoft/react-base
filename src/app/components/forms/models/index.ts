import { FieldTypes, PersistedSmartForm, SmartFormState } from '../types';

export const TEXT: FieldTypes = 'text';
export const EMAIL: FieldTypes = 'email';
export const NUMBER: FieldTypes = 'number';
export const PASSWORD: FieldTypes = 'password';
export const TEXTAREA: FieldTypes = 'textarea';
export const SWITCH: FieldTypes = 'switch';
export const CHECKBOX: FieldTypes = 'checkbox';
export const DROPDOWN: FieldTypes = 'dropdown';
export const DATE: FieldTypes = 'date';
export const BUTTON: FieldTypes = 'button';
export const SUBMIT: FieldTypes = 'submit';
export const ARRAY: FieldTypes = 'array';

const initForm = (): PersistedSmartForm => {
  return {
    state: 'pristine',
    errors: [],
  };
};

export const getState = (): SmartFormState => ({
  form: initForm(),
});
