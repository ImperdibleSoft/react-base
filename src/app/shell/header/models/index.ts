import { HeaderMenu } from '../types';

export const noMenu: HeaderMenu = '';
export const modulesMenu: HeaderMenu = 'modules';
export const companiesMenu: HeaderMenu = 'companies';
export const userMenu: HeaderMenu = 'user';
