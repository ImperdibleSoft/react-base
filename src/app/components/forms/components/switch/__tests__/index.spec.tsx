import * as React from 'react';
import { shallow } from 'enzyme';

import Switch from '..';
import { SWITCH } from '../../../models';

describe('Switch', () => {
  it('should render a valid switch field', () => {
    const options = {
      id: 'switch-field',
      label: 'Field label',
      type: SWITCH,
    };

    const component = shallow(<Switch onChange={() => {}} onClick={() => {}} onFocus={() => {}} options={options} />);

    expect(component).toMatchSnapshot();
  });
});
