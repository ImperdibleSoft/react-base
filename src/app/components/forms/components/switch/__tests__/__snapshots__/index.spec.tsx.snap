// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should render a valid switch field 1`] = `
<Wrapper>
  <Label
    htmlFor="switch-field"
  >
    <HiddenInput
      id="switch-field"
      name="switch-field"
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
    <Description>
      <CustomSwitch />
      <Content
        options={
          Object {
            "id": "switch-field",
            "label": "Field label",
            "type": "switch",
          }
        }
      />
    </Description>
  </Label>
</Wrapper>
`;
